shell/platform/darwin/ios/BUILD.gn

```
@@ -198,18 +198,20 @@ platform_frameworks_path =
 shared_library("ios_test_flutter") {
   testonly = true
   visibility = [ "*" ]
+
+  # XCode 15 beta has a bug where iOS 17 API usage is not guarded.
+  # This bug results engine build failure since the engine treats warnings as errors.
+  # The `-Wno-unguarded-availability-new` can be removed when the XCode bug is fixed.
```

Since you're here....
```suggestion
  # Xcode 15 beta has a bug where iOS 17 API usage is not guarded.
  # This bug results engine build failure since the engine treats warnings as errors.
  # The `-Wno-unguarded-availability-new` can be removed when the Xcode bug is fixed.
```

---

I'm ashamed for not having noticed this. I'll go sit in the corner for 10 minutes.